diff --git a/src/qgraphicsmozview_p.cpp b/src/qgraphicsmozview_p.cpp
index 988222d..5436db7 100644
--- a/src/qgraphicsmozview_p.cpp
+++ b/src/qgraphicsmozview_p.cpp
@@ -74,6 +74,7 @@ QGraphicsMozViewPrivate::QGraphicsMozViewPrivate(IMozQViewIface* aViewIface)
     , mIsLoading(false)
     , mLastIsGoodRotation(true)
     , mIsPasswordField(false)
+    , mIsTextArea(false)
     , mGraphicsViewAssigned(false)
     , mContentRect(0.0, 0.0, 0.0, 0.0)
     , mScrollableSize(0.0, 0.0)
@@ -397,10 +398,28 @@ void QGraphicsMozViewPrivate::IMENotification(int aIstate, bool aOpen, int aCaus
                                               const char16_t* inputType, const char16_t* inputMode)
 {
     Qt::InputMethodHints hints = Qt::ImhNone;
+    mIsPasswordField = false;
+    mIsTextArea = false;
     hints = aIstate == 2 ? Qt::ImhHiddenText : Qt::ImhPreferLowercase;
 
     QString imType((QChar*)inputType);
-    if (imType.contains("number", Qt::CaseInsensitive)) {
+    // Handle "textarea" first so that "text" is fine as well.
+    if (imType.contains("textarea", Qt::CaseInsensitive)) {
+        hints = Qt::ImhNone;
+        mIsTextArea = true;
+    }
+    else if (imType.contains("text", Qt::CaseInsensitive) ||
+        imType.contains("search", Qt::CaseInsensitive)) {
+        hints |= Qt::ImhNoAutoUppercase | Qt::ImhNoPredictiveText;
+    }
+    else if (imType.contains("password", Qt::CaseInsensitive)) {
+        hints |= Qt::ImhHiddenText;
+        hints |= Qt::ImhNoAutoUppercase;
+        hints |= Qt::ImhNoPredictiveText;
+        hints |= Qt::ImhSensitiveData;
+        mIsPasswordField = true;
+    }
+    else if (imType.contains("number", Qt::CaseInsensitive)) {
         //hints |= Qt::ImhDigitsOnly;
         hints |= Qt::ImhFormattedNumbersOnly;
     }
diff --git a/src/qgraphicsmozview_p.h b/src/qgraphicsmozview_p.h
index fa76366..7bd4e6c 100644
--- a/src/qgraphicsmozview_p.h
+++ b/src/qgraphicsmozview_p.h
@@ -112,6 +112,7 @@ public:
     bool mIsLoading;
     bool mLastIsGoodRotation;
     bool mIsPasswordField;
+    bool mIsTextArea;
     bool mGraphicsViewAssigned;
     QRectF mContentRect;
     QSizeF mScrollableSize;
diff --git a/src/quickmozview.cpp b/src/quickmozview.cpp
index b0af47f..fa1bbf3 100644
--- a/src/quickmozview.cpp
+++ b/src/quickmozview.cpp
@@ -388,12 +388,20 @@ void QuickMozView::keyPressEvent(QKeyEvent* event)
     int32_t gmodifiers = MozKey::QtModifierToDOMModifier(event->modifiers());
     int32_t domKeyCode = MozKey::QtKeyCodeToDOMKeyCode(event->key(), event->modifiers());
     int32_t charCode = 0;
+
     if (event->text().length() && event->text()[0].isPrint()) {
         charCode = (int32_t)event->text()[0].unicode();
         if (getenv("USE_TEXT_EVENTS")) {
             return;
         }
     }
+
+    if (!d->mIsTextArea && !d->mIsPasswordField &&
+        (event->key() == Qt::Key_Return || event->key() == Qt::Key_Enter) &&
+        d->mLocation.contains("youtube.com", Qt::CaseInsensitive)) {
+        charCode = domKeyCode;
+    }
+
     d->mView->SendKeyPress(domKeyCode, gmodifiers, charCode);
 }
 
@@ -405,6 +413,7 @@ void QuickMozView::keyReleaseEvent(QKeyEvent* event)
     int32_t gmodifiers = MozKey::QtModifierToDOMModifier(event->modifiers());
     int32_t domKeyCode = MozKey::QtKeyCodeToDOMKeyCode(event->key(), event->modifiers());
     int32_t charCode = 0;
+
     if (event->text().length() && event->text()[0].isPrint()) {
         charCode = (int32_t)event->text()[0].unicode();
         if (getenv("USE_TEXT_EVENTS")) {
@@ -412,6 +421,13 @@ void QuickMozView::keyReleaseEvent(QKeyEvent* event)
             return;
         }
     }
+
+    if (!d->mIsTextArea && !d->mIsPasswordField &&
+        (event->key() == Qt::Key_Return || event->key() == Qt::Key_Enter) &&
+        d->mLocation.contains("youtube.com", Qt::CaseInsensitive)) {
+        charCode = domKeyCode;
+    }
+
     d->mView->SendKeyRelease(domKeyCode, gmodifiers, charCode);
 }
 
